src/poliastro/earth/__init__.py

```diff
@@ -0,0 +1,142 @@
+
+        Parameters
+        ----------
+        r : ~astropy.units.Quantity
+            Position vector wrt attractor center.
+        v : ~astropy.units.Quantity
+            Velocity vector.
+
+        Returns
+        -------
+        Orbit
+            New 'Orbit' position and velocity vectors.
+
+
+        """
+
+        orbit = Orbit.from_vectors(Earth, r, v)
+        return cls(orbit)
+
+    @classmethod
+    @u.quantity_input(a=u.m, ecc=u.one, inc=u.rad, raan=u.rad, argp=u.rad, nu=u.rad)
+    def from_classical(
+        cls, a, ecc, inc, raan, argp, nu,
+    ):
+        """Return ` Orbit` from classical orbital elements.
+
+        Parameters
+        ----------
+        a : ~astropy.units.Quantity
+            Semi-major axis.
+        ecc : ~astropy.units.Quantity
+            Eccentricity.
+        inc : ~astropy.units.Quantity
+            Inclination
+        raan : ~astropy.units.Quantity
+            Right ascension of the ascending node.
+        argp : ~astropy.units.Quantity
+            Argument of the pericenter.
+        nu : ~astropy.units.Quantity
+            True anomaly.
+
+        Returns
+        -------
+        Orbit
+            New 'Orbit' from classical orbital elements.
+
+        """
+
+        orbit = Orbit.from_classical(Earth, a, ecc, inc, raan, argp, nu)
+        return cls(orbit)
+
+    def propagate(self, tof, *, atmosphere=None, perturbation=None):
+        """Propagates an 'EarthSatellite Orbit' at a specified time.
+
+        If value is true anomaly, propagate orbit to this anomaly and return the result.
+        Otherwise, if time is provided, propagate this `EarthSatellite Orbit` some `time` and return the result.
+
+        Parameters
+        ----------
+
+        tof : ~astropy.units.Quantity, ~astropy.time.Time, ~astropy.time.TimeDelta
+            Scalar time to propagate.
+        *:
+            parameters used in perturbation models.
+        atmosphere:
+            a callable model from poliastro.atmosphere
+        perturbation: EarthSatellitePerturbations
+            There are two possible values, SPHERICAL and J2. Only J2 is implemented at the moment. Default value is None.
+
+        Returns
+        -------
+        Orbit
+            New 'Orbit' after propagation.
+
+        """
+
+        if perturbation and perturbation.name == "J2":
+            if atmosphere:
+                a_d = partial(atmospheric_drag_model, model=atmosphere)
+                return self.orbit.propagate(
+                    tof,
+                    method=cowell,
+                    a_d=a_d,
+                    ad=J2_perturbation,
+                    J2=Earth.J2.value,
+                    R=Earth.R.to(u.km).value,
+                )
+            return self.orbit.propagate(
+                tof,
+                method=cowell,
+                ad=J2_perturbation,
+                J2=Earth.J2.value,
+                R=Earth.R.to(u.km).value,
+            )
```

Comment:
I propose that we rework the logic of this by starting with an empty `kwargs = {}` and an empty list of perturbation functions, and then adding stuff if we got propagations. Something like:

```python
kwargs = {}
perturbations = []
if gravity is EarthGravity.J2:
    kwargs.update(J2=Earth.J2.value, ...)
    perturbations.append(J2_perturbation)

if atmosphere is not None:
    kwargs.update(...)
    perturbations.append(partial(atmospheric_drag_model), model=atmosphere)

```